docs-generation/scripts/sync_frontend_translations.py

```
@@ -20,19 +30,80 @@
+                return json.load(f)
+        except Exception as e:
+            print(f"❌ Failed to read language file for {lang}: {e}")
+            return {}
+
+    # Fetch latest default branch so the checkout is up to date before reading.
+    # The upstream default branch may be main or master; try both.
+    import subprocess
+    fetched_refs = []
+    for branch in ("main", "master"):
+        try:
+            r = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)